samza-core/src/main/java/org/apache/samza/table/remote/RemoteTableProvider.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.table.remote;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.samza.SamzaException;
+import org.apache.samza.config.JavaTableConfig;
+import org.apache.samza.table.Table;
+import org.apache.samza.table.TableProvider;
+import org.apache.samza.table.TableSpec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Provide for remote table instances
+ */
+public class RemoteTableProvider implements TableProvider {
+  private static final Logger LOG = LoggerFactory.getLogger(RemoteTableProvider.class);
+
+  private final TableSpec tableSpec;
+  private RecordReader<?, ?> recordReader;
+  private RecordWriter<?, ?> recordWriter;
+
+  public RemoteTableProvider(TableSpec tableSpec) {
+    this.tableSpec = tableSpec;
+  }
+
+  @Override
+  public void start() {
```

Comment:
I am thinking since we already have init() and close(), they can probably serve the same purposes as start()/stop(). Adding another set of life-cycle methods might be too confusing.